haystack/preview/rest_api/routers/files.py

```diff
@@ -0,0 +1,88 @@
+import os
```

Comment:
From here onwards, I am unable to understand the design choices. I will write a single long comment or we can also discuss over call.